lib/services/beanstalk/deployable-artifact.js

```diff
@@ -33,62 +35,120 @@ function uploadDeployableArtifactToS3(serviceContext, fileToUpload, fileExtensio
         });
 }
 
-function prepareAndUploadDir(ownServiceContext, pathToArtifact, ebextensionsToInject) {
+function prepareAndUploadDir(accountConfig, ownServiceContext, pathToArtifact, ebextensionsToInject) {
     let tempDirPath = makeTmpDir();
 
     return util.copyDirectory(pathToArtifact, tempDirPath)
-        .then(() => {
-            ebextensions.addEbextensionsToDir(ebextensionsToInject, tempDirPath);
-            return zipDir(tempDirPath)
-                .then(fileToUpload => {
-                    let fileExtension = 'zip';
-                    return uploadDeployableArtifactToS3(ownServiceContext, fileToUpload, fileExtension)
-                        .then(s3ArtifactInfo => {
-                            //Delete temp zip file
-                            fs.unlinkSync(fileToUpload);
-                            return s3ArtifactInfo;
-                        });
-                })
-                .then(s3ArtifactInfo => {
-                    util.deleteFolderRecursive(tempDirPath); //Delete the whole temp folder
-                    return s3ArtifactInfo;
-                });
+    .then(()=>
+    {
+      // If one of the files in the directory is a Dockerrun.aws.json file then do marker substitution
+      return new Promise((resolve,reject)=>
```

Comment:
Likewise, can we refactor a bit to use synchronous file calls here? That will make this a lot more readable too, and we have the luxury of doing that since this is a CLI tool that spends most of its time waiting for AWS.